src/main/java/de/tum/cit/aet/artemis/communication/service/ReactionService.java

```diff
@@ -103,6 +117,11 @@ public void deleteReactionByIdIfAllowedElseThrow(Long reactionId, Long courseId)
         final Course course = courseRepository.findByIdElseThrow(courseId);
         Reaction reaction = reactionRepository.findByIdElseThrow(reactionId);
 
+        Course reactionCourse = getReactionCourse(reaction);
+        if (reactionCourse == null || !reactionCourse.getId().equals(courseId)) {
+            throw new BadRequestAlertException("Reaction does not belong to the given course", "reaction", "wrongCourse");
+        }
```

Comment:
Why don't we directly throw in the case if we find no course? 

I think `getReactionCourseElseThrow` would make the course simpler and also more concrete as it could make a difference if we did not find the reaction or it does not belong to the course?